src/lib/store.js

```diff
@@ -365,35 +216,48 @@ export default class Store {
 	};
 
 	selectVendor = (vendorId, isSelected) => {
-		const {selectedVendorIds} = this.vendorConsentData;
+		const {vendorConsents} = this.tcModel;
 		if (isSelected) {
-			selectedVendorIds.add(vendorId);
-		}
-		else {
-			selectedVendorIds.delete(vendorId);
+			vendorConsents.set(vendorId);
+		} else {
+			vendorConsents.unset(vendorId);
 		}
 		this.storeUpdate();
 	};
 
 	selectVendors = (vendorIds, isSelected) => {
-		const {selectedVendorIds} = this.vendorConsentData;
+		const {vendorConsents} = this.tcModel;
 		if (isSelected) {
 			vendorIds.forEach(id => {
-				selectedVendorIds.add(id);
+				vendorConsents.set(id);
 			});
-		}
-		else {
+		} else {
 			vendorIds.forEach(id => {
-				selectedVendorIds.delete(id);
+				vendorConsents.unset(id);
 			});
 		}
 		this.storeUpdate();
 	};
 
 	selectAllVendors = (isSelected) => {
-		const {vendors = []} = this.vendorList || {};
-		const operation = isSelected ? 'add' : 'delete';
-		vendors.forEach(({id}) => this.vendorConsentData.selectedVendorIds[operation](id));
+		const operation = isSelected ? 'setAllVendorConsents' : 'unsetAllVendorConsents';
+		this.tcModel[operation]();
+		this.storeUpdate();
+	};
+
+	selectVendorLegitimateInterests = (vendorId, isSelected) => {
+		const {vendorLegitimateInterests} = this.tcModel;
+		if (isSelected) {
+			vendorLegitimateInterests.set(vendorId);
+		} else {
+			vendorLegitimateInterests.unset(vendorId);
+		}
+		this.storeUpdate();
+	};
+
+	selectAllVendorLegitimateInterests = (isSelected) => {
+		const operation = isSelected ? 'setAllVendorLegitimateInterests' : 'unsetAllVendorLegitimateInterests';
+		this.tcModel[operation]();
```

Comment:
I'm not sure whether we should select legitimate interests for vendors who use no legitimate interest purposes. If there is no "Legitimate interest" checkbox in the UI for vendors who use no legitimate interest purposes we shouldn't set this flag here because user is not able to exercise right to object for such vendors.

Reply:
done